src/GraphQL/Execution/SerialExecutionStrategy.cs

```diff
@@ -75,19 +77,27 @@ await OnBeforeExecutionStepAwaitedAsync(context)
                         }
                         else if (node is IParentExecutionNode parentNode)
                         {
-                            // Add in reverse order so fields are executed in the correct order
-                            parentNode.ApplyToChildren((node, state) => state.Push(node), nodes, reverse: true);
+                            // Do not reverse the order of the nodes here
+                            parentNode.ApplyToChildren((node, state) => state.Push(node), addlNodes, reverse: false);
```

Comment:
Why `reverse` is `false` here?

Reply:
It's hard to explain.  It's like, if nodes A B are pushed in reverse order, but then under a separate node, nodes C D are pushed in reverse order, then you have B A D C instead of D C B A.  Something like that.  So the order of the data loader nodes had to be reversed immediately prior to execution of those nodes, rather than while they were being queued.

The test does test for the situation and the fix does resolve it, and it's pretty evident when looking at test output that the execution was incorrect before.

If you're still curious, I'd suggest reversing the change and running the test again to see the output.  I can't really remember all the details of this PR.